packages/lib/src/spatial-navigation/components/virtualizedList/VirtualizedList.tsx

```diff
@@ -215,14 +237,24 @@ export const VirtualizedList = typedMemo(
      * ```
      */
 
+    const getTotalVirtualizedListSize = useCallback(
+      (data: T[], itemSize: number | ((item: T) => number)) => {
+        if (typeof itemSize === 'number') {
+          return data.reduce((acc) => acc + itemSize, 0);
+        }
+        return data.reduce((acc, item) => acc + itemSize(item), 0);
+      },
+      [],
+    );
+
     const dimensionStyle = useMemo(
       () =>
         vertical
           ? ({
-              height: height + itemSize * currentlyFocusedItemIndex,
+              height: getTotalVirtualizedListSize(data, itemSize),
             } as const)
-          : ({ width: width + itemSize * currentlyFocusedItemIndex } as const),
-      [height, width, itemSize, currentlyFocusedItemIndex, vertical],
+          : ({ width: getTotalVirtualizedListSize(data, itemSize) } as const),
+      [data, getTotalVirtualizedListSize, itemSize, vertical],
```

Comment:
Since you have data here in the dependency array anyway... you can simply compute `getTotalVirtualizedListSize(data, itemSize)` once and for all in a memo with a `[data]` dependency as well.
And if the data doesn't change (supposing the consumer did memoize properly), then you win some performance.

Reply:
Done :)